x-pack/platform/plugins/shared/streams_app/public/components/data_management/schema_editor/flyout/add_field_flyout.tsx

```diff
@@ -0,0 +1,379 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiComboBoxOptionOption } from '@elastic/eui';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiTitle,
+  EuiButton,
+  EuiFormRow,
+  EuiComboBox,
+  EuiForm,
+  EuiLink,
+  useGeneratedHtmlId,
+  EuiFlyout,
+} from '@elastic/eui';
+import React, { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { isSchema, recursiveRecord, Streams } from '@kbn/streams-schema';
+import type { UseFieldsMetadataReturnType } from '@kbn/fields-metadata-plugin/public/hooks/use_fields_metadata';
+import type { SubmitHandler } from 'react-hook-form';
+import { FormProvider, useController, useForm, useFormContext, useWatch } from 'react-hook-form';
+import { CodeEditor } from '@kbn/code-editor';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { FieldMetadata } from '@kbn/fields-metadata-plugin/common';
+import { mapKeys } from 'lodash';
+import { prefixOTelField } from '@kbn/otel-semantic-conventions';
+import { useBoolean } from '@kbn/react-hooks';
+import { useKibana } from '../../../../hooks/use_kibana';
+import type { MappedSchemaField, SchemaField } from '../types';
+import { useSchemaEditorContext } from '../schema_editor_context';
+import { FieldTypeSelector as FieldTypeSelectorComponent } from './field_form_type';
+import { FieldFormFormat, typeSupportsFormat } from './field_form_format';
+import { deserializeJson, serializeXJson } from '../../stream_detail_enrichment/helpers';
+
+export interface AddFieldFlyoutProps {
+  onClose: () => void;
+  onAddField: (field: SchemaField) => void;
+}
+
+export const AddFieldButton = ({ onAddField }: Pick<AddFieldFlyoutProps, 'onAddField'>) => {
+  const [isFlyoutVisible, { on: openFlyout, off: closeFlyout }] = useBoolean(false);
+
+  return (
+    <>
+      <EuiButton
+        data-test-subj="streamsAppContentAddFieldButton"
+        iconType="plus"
+        onClick={openFlyout}
+        fill
+      >
+        {i18n.translate('xpack.streams.schemaEditor.addFieldButtonLabel', {
+          defaultMessage: 'Add field',
+        })}
+      </EuiButton>
+      {isFlyoutVisible && <AddFieldFlyout onClose={closeFlyout} onAddField={onAddField} />}
+    </>
+  );
+};
+
+export const AddFieldFlyout = ({ onAddField, onClose }: AddFieldFlyoutProps) => {
+  const { useFieldsMetadata } = useKibana().dependencies.start.fieldsMetadata;
+
+  const { stream } = useSchemaEditorContext();
+
+  const flyoutId = useGeneratedHtmlId({ prefix: 'streams-add-field' });
+
+  const { fieldsMetadata: rawFieldsMetadata } = useFieldsMetadata({
+    attributes: ['ignore_above', 'type', 'otel_equivalent'],
+    source: ['ecs', 'otel'],
+  });
```

Comment:
Get all ecs/otel fields, selecting only required properties to reduce payload size and populate the suggestions selector.